hphp/runtime/vm/jit/vasm-unit.h

```diff
@@ -211,6 +211,7 @@ struct Vunit {
   uint16_t cur_voff{0};  // current instruction index managed by Vout
   bool padding{false};
   bool profiling{false};
+  bool avoidVreg{false}; // Indication to lowerers to avoid use of new Vregs.
```

Comment:
`lowercase_with_underscores` to match the other members.

This name is also kind of misleading, because it's actually a correctness condition, not a hint.  How about `allow_implicit_vregs{true}`?

I also still think that enforcing it by means of assertions is a good idea.